eclass/qt5-build.eclass

```diff
@@ -720,11 +720,37 @@ qt5_tools_configure() {
 	# allow the ebuild to override what we set here
 	myqmakeargs=( "${qmakeargs[@]}" "${myqmakeargs[@]}" )
 
+	# out-of-source build quirks
+	case ${PN} in
+		qdoc)
+			# bug 676948
+			qt5_tools_oos_quirk src/qdoc qtqdoc-config.pri
+			;;
+		qtlocation)
+			# src/plugins/geoservices requires files that are only generated
+			# when qmake is run in the root directory. bug 633776
+			qt5_tools_oos_quirk src/location qtlocation-config.pri
+			;;
+		*) ;;
+	esac
+
 	mkdir -p "${QT5_BUILD_DIR}" || die
 	qt5_qmake "${QT5_BUILD_DIR}"
 	cp qttools-config.pri "${QT5_BUILD_DIR}" || die
```

Comment:
this code looks fairly similar, can we use `qt5_tools_oos_quirk` here as well? (might need to swap the arguments and make the directory optional, or make the logic in `qt5_tools_oos_quirk` a little smarter if only one argument is passed)